src/main/kotlin/com/openlattice/shuttle/pods/ShuttleServicesPod.kt

```diff
@@ -1,11 +1,396 @@
+import com.openlattice.auditing.AuditingConfiguration
+import com.openlattice.auth0.Auth0TokenProvider
+import com.openlattice.authentication.Auth0Configuration
+import com.openlattice.authorization.*
+import com.openlattice.authorization.initializers.AuthorizationInitializationDependencies
+import com.openlattice.authorization.initializers.AuthorizationInitializationTask
+import com.openlattice.authorization.mapstores.ResolvedPrincipalTreesMapLoader
+import com.openlattice.authorization.mapstores.SecurablePrincipalsMapLoader
+import com.openlattice.conductor.rpc.ConductorConfiguration
+import com.openlattice.data.ids.PostgresEntityKeyIdService
+import com.openlattice.data.storage.partitions.PartitionManager
+import com.openlattice.datastore.services.EdmService
+import com.openlattice.datastore.services.EntitySetService
+import com.openlattice.edm.PostgresEdmManager
+import com.openlattice.edm.properties.PostgresTypeManager
+import com.openlattice.edm.schemas.manager.HazelcastSchemaManager
+import com.openlattice.edm.schemas.postgres.PostgresSchemaQueryService
+import com.openlattice.ids.HazelcastIdGenerationService
+import com.openlattice.notifications.sms.PhoneNumberService
+import com.openlattice.organizations.HazelcastOrganizationService
+import com.openlattice.organizations.roles.HazelcastPrincipalService
+import com.openlattice.organizations.roles.SecurePrincipalsManager
+import com.openlattice.organizations.tasks.OrganizationsInitializationTask
+import com.openlattice.postgres.mapstores.EntityTypeMapstore
+import com.openlattice.shuttle.MissionParameters
+import com.openlattice.shuttle.IntegrationService
+import com.openlattice.shuttle.logs.Blackbox
+import com.openlattice.shuttle.mapstore.IntegrationsMapstore
+import com.openlattice.tasks.PostConstructInitializerTaskDependencies
+import com.openlattice.users.Auth0SyncInitializationTask
+import com.openlattice.users.Auth0SyncService
+import com.openlattice.users.Auth0SyncTask
+import com.openlattice.users.Auth0SyncTaskDependencies
+import com.zaxxer.hikari.HikariDataSource
+import org.jdbi.v3.core.Jdbi
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
+import org.springframework.context.annotation.Profile
+import java.io.IOException
+import javax.annotation.PostConstruct
+import javax.inject.Inject
+import com.openlattice.assembler.pods.AssemblerConfigurationPod
+import com.openlattice.organizations.tasks.OrganizationsInitializationDependencies
+import com.openlattice.shuttle.mapstore.IntegrationJobsMapstore
 
 /**
  *
  * @author Matthew Tamayo-Rios &lt;matthew@openlattice.com&gt;
  */
 @Configuration
+@Import(AssemblerConfigurationPod::class)
 class ShuttleServicesPod {
+    private val logger = LoggerFactory.getLogger(ShuttleServicesPod::class.java)
```

Comment:
My insides tell me that there's no way we need this much stuff just to add a controller and service but tracing this out is very painful :(

Reply:
99% of it is to get Principals services working to enable `ensureAdminAccess()` in the controller. But I agree it feels very bad.

Reply:
😢